Content.Server/StationEvents/Events/IonStormRule.cs

```diff
@@ -0,0 +1,234 @@
+                laws.Laws.RemoveAt(i);
+            }
+
+            // generate a new law...
+            var newLaw = GenerateLaw();
+
+            // see if the law we add will replace a random existing law or be a new glitched order one
+            if (laws.Laws.Count > 0 && RobustRandom.Prob(target.ReplaceChance))
+            {
+                var i = RobustRandom.Next(laws.Laws.Count);
+                laws.Laws[i] = new SiliconLaw()
+                {
+                    LawString = newLaw,
+                    Order = laws.Laws[i].Order
+                };
+            }
+            else
+            {
+                laws.Laws.Insert(0, new SiliconLaw()
+                {
+                    LawString = newLaw,
+                    Order = -1,
+                    LawIdentifierOverride = "#"
+                });
+            }
+
+            _adminLogger.Add(LogType.Mind, LogImpact.High, $"{ToPrettyString(ent):silicon} had its laws changed by an ion storm to {laws.LoggingString()}");
+
+            // laws unique to this silicon, dont use station laws anymore
+            EnsureComp<SiliconLawProviderComponent>(ent);
+            var ev = new IonStormLawsEvent(laws);
+            RaiseLocalEvent(ent, ref ev);
+        }
+    }
+
+    // for your own sake direct your eyes elsewhere
+    private string GenerateLaw()
+    {
+        // pick all values ahead of time to make the logic cleaner
+        var threats = Pick("Threats");
+        var objects = Pick("Objects");
+        var crew1 = Pick("Crew");
+        var crew2 = Pick("Crew");
+        var adjective = Pick("Adjectives");
+        var verb = Pick("Verbs");
+        var number = Pick("NumberBase") + " " + Pick("NumberMod");
+        var area = Pick("Areas");
+        var feeling = Pick("Feelings");
+        var must = Pick("Musts");
+        var require = Pick("Requires");
+        var action = Pick("Actions");
+        var allergy = Pick("Allergies");
+        var allergySeverity = Pick("AllergySeverities");
+        var species = Pick("Species");
+        var concept = Pick("Concepts");
+        var drink = Pick("Drinks");
+        var food = Pick("Foods");
+
+        var joined = $"{number} {adjective}";
+        // a lot of things have subjects of a threat/crew/object
+        var triple = RobustRandom.Next(0, 3) switch
+        {
+            0 => threats,
+            1 => crew1,
+            2 => objects
+        };
+        var crewAll = RobustRandom.Prob(0.5f) ? crew2 : "CREW";
+        var objectsThreats = RobustRandom.Prob(0.5f) ? objects : threats;
+        var objectsConcept = RobustRandom.Prob(0.5f) ? objects : concept;
+        // s goes ahead of require
+        // i dont think theres a way to do this in fluent
+        var (who, plural) = RobustRandom.Next(0, 5) switch
+        {
+            0 => ("YOU", false),
+            1 => ("THE STATION", true),
+            2 => ("THE CREW", true),
+            3 => ("THE " + crew2, false),
```

Comment:
locale for this

Reply:
i am gonna just put it here even when all these strings have locale it will be impossible to translate given how it uses english structures for logic which may not be a thing at all in different languages, or at the very least different